src/main/java/sparksoniq/JsoniqQueryExecutor.java

```diff
@@ -225,36 +225,34 @@ private RuntimeIterator generateRuntimeIterators(Expression expression) {
         return result;
     }
 
-    protected String runIterators(RuntimeIterator iterator, boolean indent) {
-        String actualOutput = getIteratorOutput(iterator, indent);
+    protected String runIterators(RuntimeIterator iterator) {
+        String actualOutput = getIteratorOutput(iterator);
         return actualOutput;
     }
 
-    private String getIteratorOutput(RuntimeIterator iterator, boolean indent) {
+    private String getIteratorOutput(RuntimeIterator iterator) {
         iterator.open(new DynamicContext());
         Item result = null;
         if (iterator.hasNext()) {
             result = iterator.next();
         }
-        if (result == null) {
-            return "";
-        }
         String singleOutput = result.serialize();
```

Comment:
This line will call serialize method on the null object and crash the app, (for example just put in the query "()" to return empty sequence).

I suggest keeping result==null check intact or making sure in some other way that the exception isn't thrown.